messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryUpdateEmitter.java

```diff
@@ -108,6 +115,30 @@ public boolean queryUpdateHandlerRegistered(SubscriptionQueryMessage<?, ?, ?> qu
                                                processor.replay(updateBufferSize).autoConnect());
     }
 
+    @Override
+    public <U> UpdateHandlerRegistration<U> registerUpdateHandler(
+            SubscriptionQueryMessage<?, ?, ?> query,
+            int updateBufferSize) {
+        Sinks.Many<SubscriptionQueryUpdateMessage<U>> sink = Sinks.many().replay().limit(updateBufferSize);
+
+        Runnable removeHandler = () -> updateHandlers.remove(query);
+        Flux<SubscriptionQueryUpdateMessage<U>> updateMessageFlux = sink.asFlux()
+                                                                        .doOnCancel(removeHandler)
+                                                                        .doOnTerminate(removeHandler);
+
+        SinksManyWrapper<SubscriptionQueryUpdateMessage<U>> sinksManyWrapper = new SinksManyWrapper<>(sink);
+
+        updateHandlers.put(query, sinksManyWrapper);
+
+        Registration registration = () -> {
+            sinksManyWrapper.complete();
```

Comment:
The underlying `Sinks.Many.complete` returns completion results. We should check on those and decide when cancelling the registration succeeds and when not.

Reply:
added `fluxSink.tryEmitComplete().orThrow()`, we use exceptions in current code